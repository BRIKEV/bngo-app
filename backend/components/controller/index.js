const System = require('systemic');
const initExpress = require('./initController');

module.exports = new System({ name: 'controller' })
  .add('controller', initExpress()).dependsOn('config', 'logger', 'store');
