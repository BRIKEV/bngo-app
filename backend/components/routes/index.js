const System = require('systemic');
const adminRoutes = require('./admin-routes');
const apiRoutes = require('./api-routes');

module.exports = new System({ name: 'routes' })
  .add('routes.admin', adminRoutes())
  .dependsOn('config', 'logger', 'server', 'middleware.prepper', 'manifest')
  .add('routes.api', apiRoutes())
  .dependsOn('logger', 'server', 'middleware.prepper')
  .add('routes')
  .dependsOn('routes.admin', 'routes.api');
