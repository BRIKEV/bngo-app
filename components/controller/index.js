const System = require('systemic');
const initExpress = require('./initController');
const imageController = require('./imageController');

module.exports = new System({ name: 'controller' })
  .add('controller', initExpress()).dependsOn('config', 'logger', 'store')
  .add('imageController', imageController())
  .dependsOn('config');
