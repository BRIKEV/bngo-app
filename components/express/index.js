const System = require('systemic');
const initExpress = require('./initExpress');

module.exports = new System({ name: 'express' })
  .add('server', initExpress()).dependsOn('config', 'logger');
