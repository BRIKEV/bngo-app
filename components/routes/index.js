const System = require('systemic');
const adminRoutes = require('./admin-routes');
const apiRoutes = require('./api-routes');

module.exports = new System({ name: 'routes' })
  .add('routes.admin', adminRoutes())
  .dependsOn('config', 'logger', 'server', 'imageController', 'middleware.prepper', 'manifest')
  .add('routes.api', apiRoutes())
  .dependsOn('logger', 'server', 'middleware.prepper', 'controller', 'config')
  .add('routes')
  .dependsOn('routes.admin', 'routes.api');
