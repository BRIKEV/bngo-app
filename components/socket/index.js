const System = require('systemic');
const initExpress = require('./initSocket');

module.exports = new System({ name: 'io' })
  .add('io', initExpress()).dependsOn('config', 'logger', 'controller', 'server');
