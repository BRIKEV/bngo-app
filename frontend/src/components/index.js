export { default as Board } from './Board/Board.vue';
export { default as ImageCard } from './ImageCard/ImageCard.vue';
export { default as OutputImages } from './OutputImages/OutputImages.vue';
