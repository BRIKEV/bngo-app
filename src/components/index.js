export { default as Board } from './Board/Board.vue';
export { default as ImageCard } from './ImageCard/ImageCard.vue';
export { default as Wheel } from './Wheel/Wheel.vue';
export { default as Notification } from './Notification/Notification.vue';
export { default as Canvas } from './Canvas/Canvas.vue';
export { default as WinnerModal } from './WinnerModal/WinnerModal.vue';
export { default as UsersList } from './UsersList/UsersList.vue';
