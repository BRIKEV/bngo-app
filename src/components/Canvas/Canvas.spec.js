import { mount, shallowMount } from '@vue/test-utils';
import { Canvas } from '@/components';

describe('Canvas component', () => {
  const propsData = {
    height: 500,
    width: 300,
  };
  const defaultConfig = {
    propsData,
  };
  it('Should be render correctly', () => {
    const wrapper = mount(Canvas, defaultConfig);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('Should be render height and width', () => {
    const wrapper = shallowMount(Canvas, {
      propsData: {
        ...defaultConfig.propsData,
        className: 'newClass',
      },
    });
    expect(wrapper.classes('newClass')).toBe(true);
  });
});
