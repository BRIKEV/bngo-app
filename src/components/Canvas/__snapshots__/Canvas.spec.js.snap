// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Canvas component Should be render correctly 1`] = `
<canvas
  class=""
  height="500"
  width="300"
/>
`;
